core/core-impl/src/main/java/ai/timefold/solver/core/impl/solver/DefaultSolver.java

```diff
@@ -234,18 +228,52 @@ public void solvingStarted(SolverScope<Solution_> solverScope) {
         solverScope.startingNow();
         solverScope.getScoreDirector().resetCalculationCount();
         super.solvingStarted(solverScope);
-        int startingSolverCount = solverScope.getStartingSolverCount() + 1;
+        var startingSolverCount = solverScope.getStartingSolverCount() + 1;
         solverScope.setStartingSolverCount(startingSolverCount);
-        logger.info("Solving {}: time spent ({}), best score ({}), environment mode ({}), "
-                + "move thread count ({}), random ({}).",
+        registerSolverSpecificMetrics();
+        logger.info("Solving {}: time spent ({}), best score ({}), "
+                + "entity count ({}), variable count ({}), "
+                + "maximum value range size ({}), "
+                + "environment mode ({}), move thread count ({}), random ({}).",
                 (startingSolverCount == 1 ? "started" : "restarted"),
                 solverScope.calculateTimeMillisSpentUpToNow(),
                 solverScope.getBestScore(),
+                solverScope.getProblemStatistics().entityCount(),
+                solverScope.getProblemStatistics().variableCount(),
+                solverScope.getProblemStatistics().maximumValueRangeSize(),
                 environmentMode.name(),
                 moveThreadCountDescription,
                 (randomFactory != null ? randomFactory : "not fixed"));
     }
 
+    private void registerSolverSpecificMetrics() {
+        solverScope.setProblemStatistics(
+                solverScope.calculateProblemStatistics(getSolverScope().getWorkingSolution()));
+        Metrics.gauge(SolverMetric.SCORE_CALCULATION_COUNT.getMeterId(), solverScope.getMonitoringTags(),
+                solverScope, SolverScope::getScoreCalculationCount);
+        Metrics.gauge(SolverMetric.PROBLEM_ENTITY_COUNT.getMeterId(), solverScope.getMonitoringTags(),
+                solverScope.getProblemStatistics(), ProblemStatistics::entityCount);
+        Metrics.gauge(SolverMetric.PROBLEM_VARIABLE_COUNT.getMeterId(), solverScope.getMonitoringTags(),
+                solverScope.getProblemStatistics(), ProblemStatistics::variableCount);
+        Metrics.gauge(SolverMetric.PROBLEM_VALUE_COUNT.getMeterId(), solverScope.getMonitoringTags(),
+                solverScope.getProblemStatistics(), ProblemStatistics::maximumValueRangeSize);
+        solverScope.getSolverMetricSet().forEach(solverMetric -> solverMetric.register(this));
+    }
+
+    private void unregisterSolverSpecificMetrics() {
+        for (var metric : List.of(SolverMetric.SCORE_CALCULATION_COUNT,
+                SolverMetric.PROBLEM_ENTITY_COUNT,
+                SolverMetric.PROBLEM_VARIABLE_COUNT,
+                SolverMetric.PROBLEM_VALUE_COUNT)) {
```

Comment:
Shouldn't we unregister the metrics that are registered, instead of relying on a list that needs to be updated every time we add a new default metric?

This is a bug in the making.

Reply:
This is a list of all solve level Solver-specific metrics (notice the solve duration longer timer is not deregistered as it not solver specific, nor are the best solution/other metrics since they are handled via event listeners).